test/compare_results.py

```diff
@@ -3,65 +3,118 @@
 import sys
 import os
 import argparse
-from pytest import approx
+import math
 
 QUANTUM_ENERGY_IDX=4
+INTENSITY_INDICES = {
+    "einstein": 2,
+    "nu": 3,
+}
 
 def find_start_end_block(lines, blockname):
-    """Identify start and end of blocks"""
+    """Identify start and end of blocks in .chk files"""
     for i, line in enumerate(lines):
         if "Start " + blockname in line:
             start_idx = i
         elif "End " + blockname in line:
             end_idx = i
     return start_idx, end_idx
 
-def extract_quantum_block(lines):
-    """Extract only the quantum block from a chk file"""
-    idxs = find_start_end_block(lines, "Quantum")
-    temp = lines[idxs[0]+4:idxs[-1]]
-    return [line.split() for line in temp]
+def find_log_block(lines, blockname):
+    """Identify start and end of block in log file"""
+    start_idx = end_idx = None
+    for i, line in enumerate(lines):
+        if blockname in line:
+            start_idx = i
+            break
+
+     # find first "done" after we've found blockname
+    for i, line in enumerate(lines[start_idx:]):
+        if "done" in line:
+            end_idx = i + start_idx
+            break
 
-def extract_quantum_energies(block):
-    """Extract quantum energies from entire block"""
-    return [float(line[QUANTUM_ENERGY_IDX]) for line in block]
+    if start_idx is not None:
+        raise Exception(f"{blockname} not found")
+    elif end_idx is not None:
+        raise Exception(f"Could not find end of {blockname}")
```

Comment:
We can fail earlier and make it a bit more abstract (not comparing to `None`) by attaching this to the loop, e.g.:
```python
for i, line in enumerate(lines):
    if blockname in line:
        ...
        break
else:
    raise Exception(...)
```

Reply:
Failing earlier certainly but I'm going to avoid using uncommon features like for-else, just to keep it as simple as possible for Sergey to edit if he ever needs to.